runtime/strings/strings.cpp

```diff
@@ -47,6 +47,14 @@ extern "C" {
   }
 
   bool hook_STRING_eq(const string * a, const string * b) {
+    uintptr_t aint = (uintptr_t) a;
+    uintptr_t bint = (uintptr_t) b;
+    if ((aint & 3) == 3 || (bint & 3) == 3) {
```

Comment:
Not claiming that there is anything wrong here (or in other olaces with a similar check), but could you explain how/where you enforce the invariants about the leats significant bits of the block pointers?

Reply:
if the least significant bit is a 1, then the other 63 bits are reserved for various constant values that would otherwise require an allocation. In this case, if the second least significant bit is also a 1, then it is a bound variable and the most significant 32 bits contains its debruijn index, otherwise it is an object-level symbol with 0 arguments and the most significanat 32 bits contains its tag. Is there a particular place in the code you think I should put this info that you would look for it there?

Reply:
Maybe as a comment in the `include/runtime/header.h`?

Reply:
I saw the commit, thanks for adding the clarification